docs/design/frontend-state-flow.md

```
@@ -0,0 +1,484 @@
+| `JobStatus` | `pollStatus` | `Status` | **`"pending"`** |
+| `JobStatus` | `pollError` | `string \| null` | `null` |
+
+---
+
+## Feature Flows
+
+### 1. Plugins Loading
+
+**Trigger:** App mounts, PluginSelector renders
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)